Source/astcenc_vecmathlib_neon_4.h

```diff
@@ -346,6 +346,16 @@ struct vmask4
 		m = vreinterpretq_u32_s32(ms);
 	}
 
+	ASTCENC_SIMD_INLINE uint32_t operator[] (int32_t i)
+	{
+		// Workaround for how msvc handles intrinsic types
+#if defined(_MSC_VER)
+		return m.n128_u32[i];
+#else
+		return m[i];
```

Comment:
Overloading the array accessors in ACLE has some weird implementation-defined behavior around endianess if you configure a big-endian Arm system. Rather than overloading "[]" I'd prefer this to provide the same "lane<id>()" function we have for the vint and vfloat types.

Reply:
Good to know! This seems cleaner overall anyway. 

Also threw in a small fix for this warning: `astc-encoder\Source\astcenc_vecmathlib_neon_4.h(885,37): warning C4244: 'return': conversion from 'int' to 'uint16_t', possible loss of data`